circle/src/domain.rs

```diff
@@ -0,0 +1,359 @@
+use alloc::vec;
+use alloc::vec::Vec;
+
+use itertools::Itertools;
+use p3_commit::{LagrangeSelectors, PolynomialSpace};
+use p3_field::extension::{Complex, ComplexExtendable};
+use p3_field::{batch_multiplicative_inverse, AbstractField, ExtensionField, Field};
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::Matrix;
+use p3_util::{log2_ceil_usize, log2_strict_usize};
+use tracing::instrument;
+
+use crate::util::{point_to_univariate, s_p_at_p, univariate_to_point, v_0, v_n};
+
+/// Given a generator h for H and an element k, generate points in the twin coset kH u k^{-1}H.
+/// The ordering is important here, the points will generated in the following interleaved pattern:
+/// {k, k^{-1}h, kh, k^{-1}h^2, kh^2, ..., k^{-1}h^{-1}, kh^{-1}, k^{-1}}.
+/// Size controls how many of these we want to compute. It should either be |H| or |H|/2 depending on if
+/// we want simply the twiddles or the full domain. If k is has order 2|H| this is equal to cfft_domain.
+pub(crate) fn twin_coset_domain<F: ComplexExtendable>(
+    generator: Complex<F>,
+    coset_elem: Complex<F>,
+    size: usize,
+) -> impl Iterator<Item = Complex<F>> {
+    generator
+        .shifted_powers(coset_elem)
+        .interleave(generator.shifted_powers(generator * coset_elem.inverse()))
+        .take(size)
+}
+
+/*
```

Comment:
nit: I would add this as possible docs, with proper reference to the Circle STARKs paper.